cadc-test-vos/src/main/java/ca/nrc/cadc/conformance/vos/SyncPullFromVOSpacePackageTest.java

```diff
@@ -0,0 +1,180 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test case for reading data from a service (pullFromVoSpace).
+ *
+ * @author jeevesh
+ */
+public class SyncPullFromVOSpacePackageTest extends VOSTransferTest
+{
+    private static Logger log = Logger.getLogger(SyncPullFromVOSpacePackageTest.class);
+
+    static
+    {
+        Log4jInit.setLevel("ca.nrc.cadc.conformance.vos", Level.INFO);
+    }
+
+    public SyncPullFromVOSpacePackageTest()
+    {
+        super(Standards.VOSPACE_SYNC_21, Standards.VOSPACE_NODES_20);
+    }
+
+    @Test
+    public void testPullFromVOSpacePackage()
+    {
+        try
+        {
+            log.debug("testPullFromVOSpacePackage");
+
+            // Get a DataNode.
+            TestNode dataNode = getSampleDataNode();
+            dataNode.sampleNode.getProperties().add(new NodeProperty(VOS.PROPERTY_URI_CONTENTLENGTH, new Long(1024).toString()));
+            WebResponse response = put(getNodeStandardID(),dataNode.sampleNode, new NodeWriter());
+            assertEquals("PUT response code should be 200", 200, response.getResponseCode());
+
+            // Create the Transfer.
+            List<Protocol> protocols = new ArrayList<Protocol>();
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTP_GET));
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTPS_GET));
+            Transfer transfer = new Transfer(dataNode.sampleNode.getUri().getURI(), Direction.pullFromVoSpace);
+            transfer.getProtocols().addAll(protocols);
+            // Add package view for tar file
+            View packageView = new View(new URI(VOS.VIEW_PACKAGE));
+            packageView.getParameters().add(new View.Parameter(new URI(VOS.PROPERTY_URI_ACCEPT), "tar"));
+            transfer.setView(packageView);
+
+            // Start the transfer.
+            TransferResult result = doSyncTransfer(transfer);
+
+            assertEquals("direction", Direction.pullFromVoSpace, result.transfer.getDirection());
+            assertNotNull("protocols", result.transfer.getProtocols());
+
+            // Get the Job.
+            response = get(result.location);
+            assertEquals("GET of Job response code should be 200", 200, response.getResponseCode());
+
+            // Job phase should be EXECUTING.
+            JobReader jobReader = new JobReader();
+            Job job = jobReader.read(new StringReader(response.getText()));
+            assertEquals("Job phase should be EXECUTING", ExecutionPhase.PENDING, job.getExecutionPhase());
+
+            // Invalid Protocol should not be returned.
+            assertTrue("has some protocols", !result.transfer.getProtocols().isEmpty());
+            String endpoint = null;
+            for (Protocol p : result.transfer.getProtocols())
+            {
+                String expectedEndpoint = "/vault/pkg/" + job.getID();
+                endpoint = p.getEndpoint();
+                log.debug("expected endpoint: " + expectedEndpoint + ", found: " + endpoint);
```

Comment:
The redirect URL to /vault/pkg won't be in a protocol endpoint, in fact you want get the transfer details (or endpoints) at all because at this point the redirect has already been followed.  (see comment above)

Reply:
From the rework discussion of Monday/Tuesday, endpoints will be returned as part of the protocols provided.